src/mento.ts

```diff
@@ -0,0 +1,260 @@
+import { BigNumber, ContractTransaction, Signer } from 'ethers'
+import {
+  IBroker,
+  IBroker__factory,
+  IExchangeProvider,
+  IExchangeProvider__factory,
+} from '@mento-protocol/mento-core-ts'
+import { Provider, TransactionResponse } from '@ethersproject/providers'
+import {
+  getBrokerAddressFromRegistry,
+  getSymbolFromTokenAddress,
+} from './utils'
+
+import { strict as assert } from 'assert'
+
+interface Exchange {
+  providerAddr: string
+  id: string
+  assets: string[]
+}
+
+interface Asset {
+  address: string
+  symbol: string
+}
+
+export class Mento {
+  private readonly provider: Provider
+  private readonly signer?: Signer
+  private readonly broker: IBroker
+  private exchanges: Exchange[]
+
+  /**
+   * This constructor is private, use the static create method to create a new Mento instance
+   * @param brokerAddress the address of the broker contract
+   * @param provider an ethers provider
+   * @param signer an optional ethers signer to execute swaps (must be connected to a provider)
+   */
+  private constructor(
+    brokerAddress: string,
+    provider: Provider,
+    signer?: Signer
+  ) {
+    this.broker = IBroker__factory.connect(brokerAddress, provider)
+    this.provider = provider
+    this.signer = signer
+
+    this.exchanges = new Array<Exchange>()
+  }
+
+  /**
+   * Create a new Mento object instance
+   * @param ethersProvider an ethers provider
+   * @param ethersSigner an optional ethers signer to execute swaps (must be connected to a provider)
+   * @returns a new Mento object instance
+   */
+  static async create(ethersProvider: Provider, ethersSigner?: Signer) {
```

Comment:
Optional: it's somewhat common to see `providerOrSigner: Provider | Signer` in methods like this, since often Signers already have connected providers in them